src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -411,6 +411,22 @@ private BoundExpression BindSimpleBinaryOperator(BinaryExpressionSyntax node, Di
             int compoundStringLength = 0;
 
             BoundExpression result = BindExpression(current, diagnostics);
+
+            if (node.IsKind(SyntaxKind.SubtractExpression))
+            {
+                if (result.Kind == BoundKind.TypeExpression
+                    && !((current as ParenthesizedExpressionSyntax)?.Expression is ParenthesizedExpressionSyntax))
+                {
+                    Error(diagnostics, ErrorCode.ERR_PossibleBadNegCast, node);
+                }
+                else if (result.Kind == BoundKind.BadExpression)
+                {
+                    var identifierName = (current as ParenthesizedExpressionSyntax)?.Expression as IdentifierNameSyntax;
+                    if (identifierName != null && identifierName.Identifier.ValueText == "dynamic")
```

Comment:
It feels like here the error should be reported only when the name didn't bind to anything. We need to figure out what would be the most accurate check for this.